### ISSUE TITLE
LC0090 Cognitive Complexity diagnostic is triggered also when exactly at the threshold

### ISSUE THREAD

**Issue by rvanbekkum** [ALCops/Analyzers#181]
The LC0090 Cognitive Complexity diagnostic is also triggered when the metric is exactly at the threshold, but the diagnostic message says "Cognitive Complexity: 30 (threshold ≤ 30)" with `≤` = 'smaller than or equal to'.

<img width="699" height="75" alt="Image" src="https://github.com/user-attachments/assets/9b57ec31-c817-4963-b81b-1a259054bc67" />

I would expect that the diagnostic would not be raised, _or_ that the diagnostic message is changed to use `<` (smaller than).

**Comment by Arthurvdv** [comment]
You’re honestly making me second-guess this. I do agree that it might feel more logical if the message only appears once the threshold is exceeded. However, I now see that in the original rule in LinterCop we already handle it this way it seems;

```Cognitive Complexity: {0} (threshold >={1})```
https://github.com/StefanMaron/BusinessCentral.LinterCop/blob/d348b710c8b15ab3f483269cb9f8c9a862d687eb/BusinessCentral.LinterCop/LinterCopAnalyzers.resx#L940

I just can’t remember the original reason behind that choice. I’ll need to check whether I can still find the best practice or guidance we based this on.

**Comment by rvanbekkum** [comment]
As you point out, the diagnostic message was different:
> Cognitive Complexity: {0} (threshold >={1})
> https://github.com/StefanMaron/BusinessCentral.LinterCop/blob/d348b710c8b15ab3f483269cb9f8c9a862d687eb/BusinessCentral.LinterCop/LinterCopAnalyzers.resx#L940

In the 'old' LinterCop it said 'threshold larger than or equal to' (makes sense to me).
In the new one it says 'threshold lower than or equal to'

The new one definitively doesn't make sense.

I think the best solution would be: change the diagnostic message to say something with 'larger than or equal to' again 😉
That makes most sense to me: 'threshold' would then mean the minimum value at which the diagnostic is triggered

**Comment by Arthurvdv** [comment]
Ah, I've was looking in the wrong direction as I didn't read your issue correctly. Thanks for pointing this out, it's indeed the incorrect threshold symbol.

**Comment by Arthurvdv** [comment]
Hi @rvanbekkum 👋

The bugfix for this issue is now available in the **beta release of ALCops 0.7.0-beta.1**. The off-by-one comparison for LC0090 has been corrected so the diagnostic no longer fires when the metric is exactly at the threshold.

We'd really appreciate it if you could try it out and let us know if the fix resolves the issue on your end. We're working towards the stable release of v0.7.0 in the upcoming days, so any feedback is very welcome!

**Comment by rvanbekkum** [comment]
I think your comment is AI-generated? Looking at the commit the diagnostic's text message was changed, not when the diagnostic is raised, right?

**Comment by Arthurvdv** [comment]
Oops, I made a copy/paste error on (my generated AI) message 🙈

Jep, the diagnostic's text message was changed, some more details in the #196.

**Comment by Arthurvdv** [comment]
With the release of version `v0.8.0` of ALCops the diagnostic message is now in the latest release version.
